platform/api/equinixmetal/api.go

```diff
@@ -158,21 +173,74 @@ func New(opts *Options) (*API, error) {
 		opts.ImageURL = defaultImageURL[opts.Board]
 	}
 
-	gapi, err := gcloud.New(opts.GSOptions)
+	url, err := url.Parse(opts.StorageURL)
 	if err != nil {
-		return nil, fmt.Errorf("connecting to Google Storage: %v", err)
+		return nil, fmt.Errorf("parsing storage URL: %w", err)
 	}
-	bucket, err := storage.NewBucket(gapi.Client(), opts.StorageURL)
-	if err != nil {
-		return nil, fmt.Errorf("connecting to Google Storage bucket: %v", err)
+
+	var storage storage.Storage
+
+	switch url.Scheme {
+	case "gs":
+		gapi, err := gcloud.New(opts.GSOptions)
+		if err != nil {
+			return nil, fmt.Errorf("connecting to Google Storage: %w", err)
+		}
+
+		bucket, err := ms.NewBucket(gapi.Client(), opts.StorageURL)
+		if err != nil {
+			return nil, fmt.Errorf("connecting to Google Storage bucket: %w", err)
+		}
+
+		storage = gcs.NewGCS(bucket)
+	case "ssh+http", "ssh+https", "ssh":
+		key, err := ioutil.ReadFile(opts.RemoteSSHPrivateKeyPath)
+		if err != nil {
+			return nil, fmt.Errorf("reading private key: %w", err)
+		}
+
+		signer, err := ssh.ParsePrivateKey(key)
+		if err != nil {
+			return nil, fmt.Errorf("parsing private key: %w", err)
+		}
+
+		cfg := &ssh.ClientConfig{
+			User: opts.RemoteUser,
+			// this is only used for testing - it's ok to live with that.
+			HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+			Auth: []ssh.AuthMethod{
+				ssh.PublicKeys(signer),
+			},
+		}
+
+		client, err := ssh.Dial("tcp", url.Host, cfg)
```

Comment:
This client establishes a TCP connection but I don't see when it gets closed. The client gets passed to the `storage` struct but it doesn't have a method to close the client. I would add one and maybe do a no-op for GCS. The only problem is when to call it, because I don't see a destructor for the API struct.

Reply:
`platform/machine/equinixmetal/flight.go` has a `Destroy` function which could call a destroy function for the API struct but it doesn't exist?